server/src/controllers/data_sources.controller.ts

```diff
@@ -74,6 +85,26 @@ export class DataSourcesController {
     return decamelizeKeys(result);
   }
 
+  @UseGuards(JwtAuthGuard)
+  @Delete(':id')
+  async delete(@Request() req, @Param() params) {
+    const dataSourceId = params.id;
+
+    const dataSource = await this.dataSourcesService.findOne(dataSourceId);
+
+    const app = await this.appsService.find(dataSource.appId);
+    const ability = await this.appsAbilityFactory.appsActions(req.user, {
+      id: app.id,
+    });
+
+    if (!ability.can('deleteQuery', dataSource.app)) {
+      throw new ForbiddenException('you do not have permissions to perform this action');
+    }
+
+    const result = await this.dataSourcesService.delete(params.id);
+    return decamelizeKeys(result);
```

Comment:
Do we need to send back postgres response back? shall we check if the operation is success from response and send back empty response with status `200`?